scene/animation/animation_tree.h

```diff
@@ -261,6 +231,116 @@ class AnimationNodeEndState : public AnimationRootNode {
 	GDCLASS(AnimationNodeEndState, AnimationRootNode);
 };
 
+// Per instance data, for a node.
+struct AnimationNodeInstance {
+	AnimationNodeInstance *parent = nullptr;
+	// TODO: Maybe ptr to parent AnimationNodeInstance for faster access???
+	AnimationNode *parent_node = nullptr;
+	// Multiple AnimationNodeInstance's can share the same node btw
+	Ref<AnimationNode> node = Ref<AnimationNode>();
+	Vector<StringName> connections; // TODO: Can we use LocalVector instead?
+	LocalVector<real_t> track_weights;
+	mutable LocalVector<int> filtered_track_indices_cache; // filtered track indices
+	StringName base_path;
+	mutable AHashMap<StringName, AnimationNodeInstance *> children_instances;
```

Comment:
```suggestion
	mutable AHashMap<StringName, AnimationNodeInstance *> child_instances;
```
No plural here in English